pkg/synchronization-controller/synchronization-controller.go

```diff
@@ -187,26 +187,33 @@ func (s *SynchronizationController) deleteMigrationFunc(delObj interface{}) {
 			return
 		}
 		if migration.Spec.Receive != nil {
-			s.closeConnectionForMigrationID(s.syncReceivingConnectionMap, migration.Spec.Receive.MigrationID)
+			if err := s.closeConnectionForMigrationID(s.syncReceivingConnectionMap, migration.Spec.Receive.MigrationID); err != nil {
+				log.Log.Reason(err).Infof("unable to close connection for migrationID %s, possibly leaked connection", migration.Spec.Receive.MigrationID)
+			}
 		} else if migration.Spec.SendTo != nil {
-			s.closeConnectionForMigrationID(s.syncOutboundConnectionMap, migration.Spec.SendTo.MigrationID)
+			if err := s.closeConnectionForMigrationID(s.syncOutboundConnectionMap, migration.Spec.SendTo.MigrationID); err != nil {
```

Comment:
I think we need to re-enqueue here. Otherwise, how can we guarantee that the connection is really closed?

Reply:
We can't, in the close connection function, we remove it from the map, so a requeue will not find it again. I could re-add it to the map, but I want to prevent an endless attempt to close that fails.

Reply:
@jean-edouard @fossedihelm what do you think?